### PR TITLE
secrets: Define Secret type

### PR DESCRIPTION
**Pull request by fishy**
This is a breaking change.

Migrate SimpleSecret and VersionedSecret from string to Secret.

Also some test improvements:

- Close tmp files after writing to make sure the writes are flushed to
  the disk
- Delete the tmp directories after tests
- Reduce the sleep time on TestSecretFileIsUpdated as we flushed the
  writes now

### CODE REVIEW

**Comment by fishy** [comment]
cc @andizzle

**Comment by andizzle** [comment]
LGTM!

File: secrets/secrets.go [hunk 1]
```diff
@@ -15,6 +15,14 @@ const (
 	credentialSecret = "credential"
 )
 
+// A Secret is the base type of secrets.
+type Secret []byte
+
+// IsEmpty returns true if the secret is empty.
+func (s Secret) IsEmpty() bool {
+	return len(s) == 0
```
Comment (andizzle) [inline secrets/secrets.go]: should we check if it's nil as well?

Comment (fishy) [reply secrets/secrets.go]: nil slice is empty slice (`len(nil)` will return 0).

Comment (andizzle) [reply secrets/secrets.go]: nope we don't need to :)

**Pull request merged**